sale_coupon_multi_gift/models/sale_coupon_program.py

```diff
@@ -0,0 +1,40 @@
+# Copyright 2021 Tecnativa - David Vidal
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from odoo import models
+
+
+class SaleCouponProgram(models.Model):
+    _inherit = "sale.coupon.program"
+
+    def _compute_order_count(self):
+        """Relay on the order line link for this programas instead of the discount
+        products"""
+        multi_gift_programs = self.filtered(lambda x: x.reward_type == "multi_gift")
+        super(SaleCouponProgram, self - multi_gift_programs)._compute_order_count()
+        for program in multi_gift_programs:
+            orders = (
+                self.env["sale.order.line"]
+                .search(
+                    [
+                        ("state", "not in", ["draft", "sent", "cancel"]),
+                        ("coupon_program_id", "=", program.id),
+                    ]
+                )
+                .mapped("order_id")
+            )
+            program.order_count = len(orders)
+
+    def action_view_sales_orders(self):
+        res = super().action_view_sales_orders()
+        if self.reward_type != "multi_gift":
+            return res
+        orders = (
```

Comment:
Cover this with a direct test of the method.

Reply:
Done